example/package.json

```diff
@@ -1,22 +1,41 @@
 {
   "name": "example",
+  "version": "0.0.1",
+  "private": true,
+  "scripts": {
+    "android": "react-native run-android --no-jetifier",
+    "ios": "react-native run-ios",
+    "start": "react-native start",
+    "test": "jest",
+    "lint": "eslint .",
+    "postinstall": "rm -rf node_modules/react-native-safe-area-context && patch-package"
+  },
   "dependencies": {
-    "@react-native-community/async-storage": "^1.10.1",
-    "@react-native-community/masked-view": "^0.1.10",
-    "@react-navigation/bottom-tabs": "^5.4.4",
-    "@react-navigation/native": "^5.3.2",
-    "@react-navigation/stack": "^5.3.6",
-    "expo": "^37.0.8",
-    "react": "^16.13.1",
-    "react-dom": "^16.13.1",
-    "react-native": "^0.63.0-rc.0",
-    "react-native-gesture-handler": "^1.6.1",
-    "react-native-reanimated": "^1.8.0",
-    "react-native-safe-area-view": "^1.1.1",
-    "react-native-screens": "^2.7.0",
-    "react-native-web": "^0.12.2",
-    "react-navigation": "^4.3.9",
-    "react-navigation-stack": "^2.5.0",
-    "react-navigation-tabs": "^2.8.13"
+    "@react-native-community/async-storage": "^1.12.1",
+    "@react-navigation/native": "^6.0.6",
+    "@react-navigation/stack": "^6.0.11",
+    "@react-navigation/bottom-tabs": "^6.0.9",
+    "react": "17.0.2",
+    "react-native": "0.0.0-20211215-2008-f16cbe59e",
+    "react-native-codegen": "^0.0.12",
+    "react-native-gradle-plugin": "^0.0.3",
+    "react-native-safe-area-context": "../"
```

Comment:
I was wondering, why it has been added here, but probably I found a explanation for that 

- the dep is added to package.json
- codegen recognize it as a dep 
- using a custom path the codegen generates needed files 
- in `postInstall` phase this dep is. removed from node modules to avoid duplication 

Could you confirm, if my above thought are correct ?

Reply:
Yep that's why, it's a hack to make codegen discover the library properly, while still using the source in the repo with the babel plugin alias to make dev simple.

Not sure if there's a better way to setup that, but for now that works.